features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -411,6 +409,7 @@ nsapi_error_t mbed_lwip_init(emac_interface_t *emac)
             return NSAPI_ERROR_DEVICE_ERROR;
         }
 
+        mbed_lwip_set_mac_address();
```

Comment:
@bulislaw It's in the if (!mbed_lwip_get_mac_address()) scope. The closing bracket above it is for the case when netif_add fails.